onnxruntime/core/providers/cpu/tensor/upsamplebase.h

```diff
@@ -58,7 +60,70 @@ enum class AspectRatioPolicy {
   NOT_SMALLER,
 };
 
+// Antialias types
+template <typename T>
+struct AccumulateType {
+  using type = int32_t;
+  using Dtype = T;
+};
+
+template <>
+struct AccumulateType<int32_t> {
+  using type = float;
+};
+
+template <>
+struct AccumulateType<float> {
+  using type = float;
+};
+
+template <>
+struct AccumulateType<MLFloat16> {
+  using type = float;
+};
+
+template <>
+struct AccumulateType<double> {
+  using type = double;
+};
+
+// Antialiasing constants
+constexpr float kCubicCoeffA = -0.75f;
+constexpr float kSupportSize = 2.0f;
+constexpr float kBiCubicSupportSize = 4.0f;
+
+namespace ConstValue {
+constexpr int32_t mag_factor = 1 << (22 - 1);
```

Comment:
This is only used once in
https://github.com/microsoft/onnxruntime/blob/a9568935a52b3d51ec802a4ab89ab3852129fc1e/onnxruntime/core/providers/cpu/tensor/upsample_antialias.h#L248

I think it is better to be defined in that function like:
```
constexpr float mag_factor = 1 << 22;
scale_buffer_int[x] = static_cast<int32_t>(std::round(scale_buffer[x] * mag_factor)); 
```